tests/codegen/enum.jakt

```diff
@@ -0,0 +1,9 @@
+/// Expect:
+/// - output: ""
+
+enum Test {
+    Some(s: String)
+    None()
```

Comment:
Can we just disallow this instead? or transparently convert this typed variant into an untyped one?

Reply:
Yeah, imo we should emit a warning and convert it to untyped.

Reply:
I don't understand, why do you want to disallow this? This is extremely useful. And its not untyped at all. This will be extremely useful as a return type for example returning an event, some events having payload, others not.

Reply:
That and it allows for a very simple and beautiful in-language optional, or variants which have a none state, i.e. shorthanding Optional<Varian>, which makes these much nicer to use imho. Please don't disallow this, I really want it :).

Reply:
@maddanio 
What we mean is that
```
enum Test {
    Some(s: String)
    None
}
```
is already supported and `None` (without the parentheses) should probably be either the only way to write it or highly encouraged to be how you write it.
Also there already is an in-language optional:
```
enum Optional<T> {
    Some(T)
    None
}
```
Every `type?` gets converted to `Optional<type>` behind the scenes :)
I'm not entirely sure what you mean with how to handle events, but maybe I already addressed that. If not I'm happy to discuss this more :D

Reply:
aaaaah, now I get it. you lost me with untyped. None is a type in my head. so yes, basically the parentheses should be forbidden, ok, will try that :-D